gen-requirements.txt

```diff
@@ -0,0 +1,6 @@
+-c dev-requirements.txt
```

Comment:
Added a separate file for two reasons:

1. tox -e generate only needs these dependencies. Installing everything else from dev-requirements.txt is very unnecessary.
2. astor pulls in deps that fails on pypy so if we add astor to dev-requirements, pypy test jobs start failing.